common/djangoapps/util/models.py

```diff
@@ -12,3 +18,53 @@ class RateLimitConfiguration(ConfigurationModel):
     with the `can_disable_rate_limit` class decorator.
     """
     pass
+
+
+def uncompress_string(s):
+    """helper function to reverse django.utils.text.compress_string"""
+
+    try:
+        val = s.encode('utf').decode('base64')
+        zbuf = cStringIO.StringIO(val)
+        zfile = gzip.GzipFile(fileobj=zbuf)
+        ret = zfile.read()
+        zfile.close()
+    except:
```

Comment:
1. Done.
2. Done.
3. I don't know. That code comes from https://djangosnippets.org/snippets/2893/. The entire method is simply a reversal of `CompressedTextField.get_prep_value`.